compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -317,6 +317,13 @@ impl<'cx, 'tcx> FallibleTypeFolder<'tcx> for QueryNormalizer<'cx, 'tcx> {
 
             _ => ty.try_super_fold_with(self),
         })()?;
+
+        // `tcx.normalize_projection_ty` may normalize to a type that still has
+        // unevaluated consts, so keep normalizing here if that's the case.
+        if res != ty && res.has_type_flags(ty::TypeFlags::HAS_CT_PROJECTION) {
+            res = res.try_super_fold_with(self)?;
```

Comment:
can you instead move this into the `ty::Projection` branches

I think that's where this should happen conceptually.

after that r=me, gj on figuring this out :+1: